vegbilder-client/src/components/StatisticsInfoBox/StatisticsInfoBox.tsx

```diff
@@ -30,71 +30,119 @@ const useStyles = makeStyles(() => ({
     const statisticsGroupedByYear = groupBy(statistics, i => i.AAR);
-    const tableRows: IStatisticsRow[] = [];
-    Object.keys(statisticsGroupedByYear).map((year) => {
+
+    const tableRows: IStatisticsRow[] = Object.keys(statisticsGroupedByYear).map((year) => {
         const row = Object.create({});
         row[`year`] = year;
-        statisticsGroupedByYear[year].map((category) => {
-            row[`${category.VEGKATEGORI}`] = category.ANTALL;
+        statisticsGroupedByYear[year].forEach((category) => {
+            if (vegvesenRoadCategories.includes(category.VEGKATEGORI)) {
+                row[`${category.VEGKATEGORI}`] = category.ANTALL;
+            } else {
+                if (!(row.hasOwnProperty(`other`))) {
+                    row[`other`] = category.ANTALL;
+                } else {
+                    row[`other`] += category.ANTALL;
+                }
+            }
         });
-        tableRows.push(row);
+        return row;
     })
     return tableRows as IStatisticsRow[];
 }
 
+
+const sortTableRowsBasedOnYear = (tableRows: IStatisticsRow[]) => {
+    return tableRows.sort((rowA, rowB) => (rowA.year < rowB.year) ? 1 : -1);
+}
+
+//Går ut ifra at inneværende å alltid sendes med selv om antall på alle veger skulle være 0.
+const getRowForCurrentYear = (sortedTableRows: IStatisticsRow[]) => {
+    return sortedTableRows[0];
+}
+
+//Synes ikke denne funskjonen er veldig pent skrevet per nå.
+const createTotalRowExcludingCurrentYear = (sortedTableRows: IStatisticsRow[]) => {
+    const currentYear = new Date().getFullYear();
+    const previousYear = currentYear - 1;
+    const rowListWithoutCurrentYear = sortedTableRows.filter((row) => row.year !== currentYear.toString());
+    const totalE = rowListWithoutCurrentYear.reduce((prev, cur) => prev + cur.E, 0);
+    const totalR = rowListWithoutCurrentYear.reduce((prev, cur) => prev + cur.R, 0);
+    const totalF = rowListWithoutCurrentYear.reduce((prev, cur) => prev + cur.F, 0);
+
+    //Legger bare sammen dersom "other" eksisterer på gjeldende rad.
+    const onlyExistingOtherValuesReducer = (prev: number, cur: IStatisticsRow) => {
+        return cur.other !== null || cur.other !== undefined ? prev + cur.other : prev;
+    }
+
+    let categoryOtherExists = false;
+    rowListWithoutCurrentYear.forEach((row) => row?.other ? categoryOtherExists = true : null);
+
+    if (categoryOtherExists) {
+        const totalOfCategoryOther = rowListWithoutCurrentYear.reduce(onlyExistingOtherValuesReducer, 0);
+        return {
+            year: `${previousYear.toString()} og eldre`,
+            E: totalE,
+            R: totalR,
+            F: totalF,
+            other: totalOfCategoryOther
+        } as IStatisticsRow
+    } else {
+        return {
+            year: `${previousYear.toString()} og eldre`,
+            E: totalE,
+            R: totalR,
+            F: totalF
+        } as IStatisticsRow;
+    }
+}
```

Comment:
Tanken her er å lage et objekt med totalt antall bilder for hver vegkategori for alle år unntatt inneværende. Må derfor summere vegkategoriene for hvert radobjekt i tableRows og putte disse i et nytt rad-objekt. Må også tenke på at det ikke nødvendigvis er et attributt "other" på en rad fordi det ikke nødvendigvis finnes bilder fra andre vegtyper i det gitte året. Har prøvd meg på en løsning, men synes ikke den er så pent skrevet. Hvis du har tid blir jeg veldig glad for forbedringsforslag @LangdalP 🙂